agenta-backend/agenta_backend/migrations/mongo_to_postgres/migration.py

```diff
@@ -0,0 +1,522 @@
+        "updated_at": get_datetime(deployment.get("updated_at")),
+    }
+
+
+async def transform_variant_base(base):
+    app_uuid = await get_mapped_uuid(base["app"].id)
+    user_uuid = await get_mapped_uuid(base["user"].id)
+    image_uuid = await get_mapped_uuid(base["image"].id)
+    deployment_uuid = base["deployment"] and await get_mapped_uuid(base["deployment"])
+    base_uuid = generate_uuid()
+    await store_mapping("bases", base["_id"], base_uuid)
+    return {
+        "id": base_uuid,
+        "app_id": app_uuid,
+        "user_id": user_uuid,
+        "base_name": base["base_name"],
+        "image_id": image_uuid,
+        "deployment_id": deployment_uuid,
+        "created_at": get_datetime(base.get("created_at")),
+        "updated_at": get_datetime(base.get("updated_at")),
+    }
+
+
+async def transform_app_variant(variant):
+    app_uuid = await get_mapped_uuid(variant["app"].id)
+    image_uuid = await get_mapped_uuid(variant["image"].id)
+    user_uuid = await get_mapped_uuid(variant["user"].id)
+    modified_by_uuid = await get_mapped_uuid(variant["modified_by"].id)
+    base_uuid = await get_mapped_uuid(variant["base"].id)
+    variant_uuid = generate_uuid()
+    await store_mapping("app_variants", variant["_id"], variant_uuid)
+    return {
+        "id": variant_uuid,
+        "app_id": app_uuid,
+        "variant_name": variant["variant_name"],
+        "revision": variant["revision"],
+        "image_id": image_uuid,
+        "user_id": user_uuid,
+        "modified_by_id": modified_by_uuid,
+        "base_name": variant.get("base_name"),
+        "base_id": base_uuid,
+        "config_name": variant["config_name"],
+        "config_parameters": variant["config"],
+        "created_at": get_datetime(variant.get("created_at")),
+        "updated_at": get_datetime(variant.get("updated_at")),
+    }
+
+
+async def transform_app_variant_revision(revision):
+    variant_uuid = await get_mapped_uuid(revision["variant"].id)
+    modified_by_uuid = await get_mapped_uuid(revision["modified_by"].id)
+    base_uuid = await get_mapped_uuid(revision["base"].id)
+    revision_uuid = generate_uuid()
+    await store_mapping("app_variant_revisions", revision["_id"], revision_uuid)
+    return {
+        "id": revision_uuid,
+        "variant_id": variant_uuid,
+        "revision": revision["revision"],
+        "modified_by_id": modified_by_uuid,
+        "base_id": base_uuid,
+        "config_name": revision["config"]["config_name"],
+        "config_parameters": revision["config"]["parameters"],
+        "created_at": get_datetime(revision["created_at"]),
+        "updated_at": get_datetime(revision["updated_at"]),
+    }
+
+
+async def transform_app_environment(environment):
+    app_uuid = await get_mapped_uuid(environment["app"].id)
+    user_uuid = await get_mapped_uuid(environment["user"].id)
+    variant_uuid = await get_mapped_uuid(environment["deployed_app_variant"])
```

Comment:
I see that sometimes you use `.id` and sometimes not. I would add some assertion on the datatype here just to prevent errors.

Reply:
this is because they differ in the way they are persisted. some are using the link some we simply persist the object id
Example: 
```
'app': DBRef('app_db', ObjectId('123')),
'user': DBRef('users', ObjectId('456')),
'deployed_app_variant': ObjectId('678'),
```